src/reactjs/components/CIRAEditor/CiraEditor.tsx

```diff
@@ -2,138 +2,138 @@
+  .map((icon) => Icons[icon])
+library.add(...iconList)
 
 export interface CiraEditorProps {
-  rpsServer: string;
-  mpsServer: string;
+  rpsServer: string | null
+  mpsServer: string | null
 }
 
 export interface CiraEditorState {
-  openFlyout?: boolean;
-  showPopup?: boolean;
-  showMessage?: boolean;
-  message?: string;
-  type?: string;
-  updateCiraGrid?: boolean;
-  selectedCiraConfigs: any;
+  openFlyout?: boolean
+  showPopup?: boolean
+  showMessage?: boolean
+  message?: string
+  type?: string
+  updateCiraGrid?: any
+  selectedCiraConfigs: any
   isEdit: boolean
 }
 
 /**
  * Wrapper component for rendering CIRA grid and header for CIRA config scripts control
  */
 export class CiraEditor extends React.Component<
-  CiraEditorProps,
-  CiraEditorState
-  > {
-  constructor(props: CiraEditorProps) {
-    super(props);
+CiraEditorProps,
+CiraEditorState
+> {
+  constructor (props: CiraEditorProps) {
+    super(props)
     this.state = {
       openFlyout: false,
       showPopup: false,
       showMessage: false,
-      message: "",
-      type: "",
+      message: '',
+      type: '',
       updateCiraGrid: false,
-      selectedCiraConfigs: "",
+      selectedCiraConfigs: '',
       isEdit: false
-    };
+    }
   }
 
-  togglePopup = () => {
+  togglePopup = (): void => {
     this.setState({
-      showPopup: !this.state.showPopup,
-    });
-  };
+      showPopup: !isFalsy(this.state.showPopup)
+    })
+  }
 
-  confirmDelete = async () => {
-    const { rpsKey } = this.context.data;
-    this.togglePopup();
-    let configName = encodeSpecialCharacters(this.state.selectedCiraConfigs[0].configName);
-    let response = await HttpClient.delete(`${this.props.rpsServer}/api/v1/admin/ciraconfigs/${configName}`, rpsKey)
-    if (response === `CIRA Config ${this.state.selectedCiraConfigs[0].configName} successfully deleted`) {
+  confirmDelete = async (): Promise<any> => {
+    const { rpsKey } = this.context.data
+    this.togglePopup()
+    const configName = encodeSpecialCharacters(this.state.selectedCiraConfigs[0].configName)
+    const response = await HttpClient.delete(`${String(this.props.rpsServer)}/api/v1/admin/ciraconfigs/${configName}`, rpsKey)
+    if (response === `CIRA Config ${String(this.state.selectedCiraConfigs[0].configName)} successfully deleted`) {
       this.setState({
         showMessage: true,
         message: response,
-        type: `success`,
-        updateCiraGrid: !this.state.updateCiraGrid,
-        selectedCiraConfigs: "",
-      });
+        type: 'success',
+        updateCiraGrid: !isFalsy(this.state.updateCiraGrid),
+        selectedCiraConfigs: ''
+      })
     } else {
       this.setState({
         showMessage: true,
         message: response,
-        type: `error`,
-      });
+        type: 'error'
+      })
     }
-    this.showNotification();
+    this.showNotification()
   }
 
-  showNotification = () => setTimeout(() => {
+  showNotification = (): any => setTimeout(() => {
```

Comment:
this should be void?

Reply:
Since setTimeout function returns a value, the return type has been updated to match the return type of setTimeout function.